sphinxcontrib/mat_types.py

```diff
@@ -1170,16 +1172,87 @@ def __doc__(self):
 
 
 class MatScript(MatObject):
-    def __init__(self, name, path, tks):
+    def __init__(self, name, modname, tks):
         super(MatScript, self).__init__(name)
-        self.path = path
-        self.tks = tks
+        #: Path of folder containing :class:`MatScript`.
+        self.module = modname
+        #: List of tokens parsed from mfile by Pygments.
+        self.tokens = tks
+        #: docstring
         self.docstring = ''
+        #: remaining tokens after main function is parsed
+        self.rem_tks = None
+
+        tks = copy(self.tokens)  # make a copy of tokens
+        tks.reverse()  # reverse in place for faster popping, stacks are LiLo
+        skip_whitespace(tks)
+        # =====================================================================
+        # docstring
+        try:
+            docstring = tks.pop()
+        except IndexError:
+            docstring = None
+        while docstring and docstring[0] is Token.Comment:
```

Comment:
Only works with when the docstring starts on the first line. I think it should work as MATLAB does, where help on a script returns the first comment block.

Reply:
The whitespace is already removed from the beginning of the file before reading the tokens for the header. I added a test to show this and updated the test suite.

It this what you were hoping for?

Reply:
Almost. In MATLAB you can have code before the comment, and it will still use the first comment as docstring:

Example `script.m` containing:

    x = 2*pi;

    % The docstring

    y = cos(x);

Typing `help script` will print

    The docstring

You're almost there.

Reply:
I fixed the behaviour and added a test for it.

I was confused because I tried on my machine and it didn't show the doctoring. I found that MATLAB help only works when there is 1 statement before the doctoring. It doesn't work if there is more than one statement.

My fix consumes the statements before the docstring, so it get to the docstring even if there is more than one statement.

Reply:
Sorry, you're absolutely right. I think this solution is good enough for releasing.